src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -17,6 +17,7 @@ internal class WinHttpResponseStream : Stream
         private volatile bool _disposed;
         private readonly WinHttpRequestState _state;
         private SafeWinHttpHandle _requestHandle;
+        private CancellationTokenRegistration _ctr;
```

Comment:
I could store this _ctr inside the _state object (there should only be one active at a time) if you think it is important enough to Dispose() right away.